File: examples/Student.java
import java.util.HashMap;

public class Student {

	private static HashMap<String, String> associations;

	public Student(String name, String student_id, String programme) {
		this.name = name;
		this.student_id = student_id;
		this.programme = programme;

		associations = new HashMap<>();
		associations.put("Applied Physics", "Arago");
		associations.put("Computer Science", "Inter-actief");
		associations.put("Chemical Engineering", "alembic");
	}

	public String getAssociation() {
		return this.associations(this.programme);
	}

}

File: examples/student.m
classdef student
  properties (Constant)
    associations = struct(...
    'Applied Physics', 'Arago', ...
    'Computer Science', 'Inter-actief', ...
    'Chemical Engineering', 'alembic');
  end
  
  methods
    % The constructor
    function obj = student(name, student_id, programme)
      obj.name = name;
      obj.student_id = student_id;
      obj.programme = programme;
    end
    
    function a = getAssociation()
        a = obj.associations(obj.programme);
    end
  end
  
end
